lib/grandpa/message_tracker.go

```diff
@@ -71,6 +72,13 @@ func (t *tracker) addCommit(cm *CommitMessage) {
 	t.commitMessages[cm.Vote.Hash] = cm
 }
 
+func (t *tracker) addCatchUpResponse(cr *CatchUpResponse) {
+	t.mapLock.Lock()
+	defer t.mapLock.Unlock()
```

Comment:
Should we have a `catchUpResponseMessages` specific mutex instead of `mapLock`? To prevent eventual deadlocks and/or reduce locking contention

Reply:
I like the idea of preventing eventual deadlocks. Right now, we process all kinds of messages (vote and commit) in the same function and for reprocessing also we have a single function. I don't see possibility of a deadlock there.

For catch up messages, I am yet not sure when are we going to reprocess them. I am going to change this map to sync.Map and add a todo for it to revisit in https://github.com/ChainSafe/gossamer/issues/1531